lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4524,3 +4572,20 @@ DynamicallyReplacedDeclRequest::evaluate(Evaluator &evaluator,
 
   return nullptr;
 }
+
+void TypeChecker::typeCheckDerivativeAttrs(SourceFile &sourceFile) {
+  class DerivativeAttrBindingWalker : public ASTWalker {
+    bool walkToDeclPre(Decl *decl) override {
+      auto f = dyn_cast<AbstractFunctionDecl>(decl);
+      if (!f)
+        return true;
+      for (auto *attr : f->getAttrs())
```

Comment:
This doesn't easily work because `.getAttributes<DerivativeAttr>()` returns a const iterator, but typeCheckDerivativeAttr mutates the attribute (`attr->setParameterIndices()`, `attr->setInvalid()`).

The invocation here doesn't really need to mutate the attribute, so typeCheckDerivativeAttr could be refactored into a mutating and nonmutating part, though that would make this change bigger.

Reply:
Got it, thanks!